File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.HashMap;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int[] answer = new int[photo.length];
        HashMap<String, Integer> scoreMap = new HashMap<>();
        
        for (int i = 0; i < name.length; i++) {
            scoreMap.put(name[i], yearning[i]);
        }
        
        for (int i = 0; i < photo.length; i++) {
            int currentScore = 0;
            
            for (String person : photo[i]) {
                currentScore += scoreMap.getOrDefault(person, 0);
            }
            answer[i] = currentScore;
        }
        return answer;
    }
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

vector<int> solution(vector<string> name, vector<int> yearning, vector<vector<string>> photo) {
    map<string, int> score_map;
    vector<int> answer;
    
    for (int i = 0; i < name.size(); i++) {
        score_map.insert({name[i], yearning[i]});
    }
    
    for (int i = 0; i < photo.size(); i++) {
        int sum = 0; 
        for (int j = 0; j < photo[i].size(); j++) {
            string person = photo[i][j];
            
            if (score_map.find(person) != score_map.end()) {
                sum += score_map[person];
            }
        }
        answer.push_back(sum);
    }
    return answer;
}